pkg/parse/run_test.go

```diff
@@ -236,48 +236,62 @@ func TestReconciler_Reconcile(t *testing.T) {
 		expectedSourceChanged bool
 		needRetry             bool
 		parseOutputs          []fsfake.ParserOutputs
-		expectedRootSync      *v1beta1.RootSync
+		expectedRootSyncFunc  func(sourcePath string) *v1beta1.RootSync
 	}{
 		{
-			name:                  "read retryable error because missing source commit directory",
-			retryCap:              5 * time.Millisecond,
-			srcRootCreateLatency:  10 * time.Millisecond,
-			expectedSourceChanged: false,
-			needRetry:             true,
-			parseOutputs:          nil, // parse should not be called
-			expectedRootSync: func() *v1beta1.RootSync {
+			name:                  "reconcile success with existing source commit directory",
```

Comment:
I moved the `read retryable error` test down a bit, which broke the expectations, which were apparently hard-coded based on the test order. So I had to change `expectedRootSync`  to `expectedRootSyncFunc`, so I could pass in the sourcePath. 

I also added a new `reconcile success with existing source commit directory` test happy path test.
That's why this diff is a little jumbled.

Reply:
I also added a new `reconcile success with full-sync after apply error` test, which needed `reconcilerStateFunc` to initialize the state to simulate a previous apply error on the same commit.